pkg/agent/core/ngt/handler/grpc/handler.go

```diff
@@ -443,20 +446,60 @@ func (s *server) Insert(ctx context.Context, req *payload.Insert_Request) (res *
 	vec := req.GetVector()
 	err = s.ngt.Insert(vec.GetId(), vec.GetVector())
 	if err != nil {
-		err = status.WrapWithInternal("Insert API failed", err,
-			&errdetails.RequestInfo{
-				RequestId:   req.GetVector().GetId(),
-				ServingData: errdetails.Serialize(req),
-			},
-			&errdetails.ResourceInfo{
-				ResourceType: ngtResourceType + "/ngt.Insert",
-				ResourceName: s.ip,
-				Owner:        errdetails.ValdResourceOwner,
-				Description:  err.Error(),
-			}, info.Get())
-		log.Error(err)
+		var code trace.Status
+		if errors.Is(err, errors.ErrUUIDAlreadyExists(vec.GetId())) {
+			err = status.WrapWithAlreadyExists(fmt.Sprintf("Insert API uuid %s already exists", vec.GetId()), err,
+				&errdetails.RequestInfo{
+					RequestId:   req.GetVector().GetId(),
+					ServingData: errdetails.Serialize(req),
+				},
+				&errdetails.ResourceInfo{
+					ResourceType: ngtResourceType + "/ngt.Insert",
+					ResourceName: s.ip,
+					Owner:        errdetails.ValdResourceOwner,
+					Description:  err.Error(),
+				})
+			log.Warn(err)
+			code = trace.StatusCodeAlreadyExists(err.Error())
+		} else if errors.Is(err, errors.ErrUUIDNotFound(0)) {
+			err = status.WrapWithInvalidArgument(fmt.Sprintf("Insert API invalid uuid \"%s\" detected", vec.GetId()), err,
```

Comment:
Maybe better to tell "uuid is empty".

```suggestion
			err = status.WrapWithInvalidArgument(fmt.Sprintf("Insert API empty uuid \"%s\" was given", vec.GetId()), err,
```

Reply:
Okay